http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -415,6 +422,51 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift](
       _ <- gceRuntimeMonitor.process(runtime.id).compile.drain
     } yield ()
 
+  private[monitor] def handleCreateDiskMessage(msg: CreateDiskMessage, now: Instant)(
+    implicit traceId: ApplicativeAsk[F, AppContext]
+  ): F[Unit] =
+    for {
+      _ <- googleDiskService.createDisk(
+        msg.googleProject,
+        msg.zone,
+        Disk
+          .newBuilder()
+          .setId(msg.diskId.toString)
+          .setCreationTimestamp(now.toString)
+          .setName(msg.name.toString)
+          .setSizeGb(msg.size.gb.toString)
+          .setZone(msg.zone.toString)
+          .build()
+      )
+    } yield ()
+
+  private[monitor] def handleDeleteDiskMessage(msg: DeleteDiskMessage, now: Instant)(
+    implicit traceId: ApplicativeAsk[F, AppContext]
+  ): F[Unit] =
+    for {
+      diskOpt <- persistentDiskQuery.getById(msg.diskId).transaction
+      disk <- diskOpt.fold(
+        F.raiseError[PersistentDisk](PubsubHandleMessageError.DiskNotFound(msg.diskId, msg))
+      )(F.pure)
+      // TODO: do I need to check disk status before deleting?
```

Comment:
@akarukappadath @rtitle do I need to check disk status before deleting? and what about for updating disk? what statuses are we checking for?

Reply:
I think we expect the disk to be in `Deleting` state by the time the subscriber receives the `delete` message. So maybe check the record for `Deleting` status, and raise an error otherwise (this is similar to `handleDeleteRuntimeMessage`).